app/src/main/java/org/sopt/and/SignInActivity.kt

```diff
@@ -0,0 +1,150 @@
+package org.sopt.and
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+class SignInActivity : ComponentActivity() {
+    private var registeredId: String? = null
+    private var registeredPassword: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // 회원가입으로부터 데이터를 받기 위한 Intent
+        registeredId = intent.getStringExtra("ID")
+        registeredPassword = intent.getStringExtra("Password")
```

Comment:
"ID"과 같은 키값은, 만약 변경이 생긴다면 보내는쪽 받는쪽 모두 수정이 필요하기 때문에 한쪽만 수정되는 경우 오류가 발생하기 쉽습니다! 그래서 일반적으로 const val을 활용하여 따로 저장합니다.
일반 변수가 아닌 const val을 사용하는 이유는 무엇일까요~ 장점을 알아두면 좋을 것 같아요!

또한 이렇게 intent가 반복되는 경우 스코프 함수를 활용할 수도 있어요!
```suggestion
        with(intent) {
                registeredId = getStringExtra(KEY_ID)
                registeredPassword = getStringExtra(KEY_PW)
        }
```

Reply:
네 알겠습니다! 양질의 피드백 감사합니다ㅎㅎ

Reply:
찾아보니 const val은 불변성과 효율성, 성능최적화, 유지보수성, 안정성이 장점이고 이는 const val이 컴파일 시간 상수이기 때문이라고 합니다!

Reply:
ㅎㅎ맞아요 이 경우에는 
companion object {
    const val KEY_ID = ~
}
처럼 활용하는 것이 좋은데, companion object 에 대해서도 한번 알아보시겠어요?

Reply:
맞아요! 다만 저는 Java를 몰라서 와닿지 않았었는데요...ㅎㅎㅎ
companion object는 공유 데이터나 공용 메서드를 한 곳에서 관리할 수 있어, 이를 다른 액티비티나 클래스에서 쉽게 접근하고 재사용할 수 있는 방법을 제공합니다!

```
putExtra(KEY_ID, "값")
companion object {
    const val KEY_ID = ~
}
```
처럼 FirstActivity에서 키값을 저장했었다면

```
getStringExtra(FirstActivity.KEY_ID)
```

로 값을 가져올 수 있어요! 그러면 const val KEY_ID 을 이후에 변경해도 동시에 변경되겠죠?

Reply:
넘 딱딱하게 안해도 돼요 ㅎㅎㅎㅎ 더 궁금한거 있으시면 편하게 연락주십쇼 !